### ISSUE TITLE
CI for Fastlane + Android

### ISSUE THREAD

**Issue by olyism** [buildkite/templates#43]

**Comment by olyism** [comment]
Here's a related comment from @sj26 

> Hi folks! Do we have an example pipeline for building an android app? Even better, do we have something for building a library for android? I want some example pipelines I can rig up that build and push packages from within pipelines.

I'm unsure if Fastlane is the right workflow tool for this. Leaving a note here to investigate later.

**Comment by sj26** [comment]
Fastlane is a nice value-add for Android builds, as part of the continuous delivery story. But there should probably be an example using [Gradle](https://developer.android.com/build) directly as well. Getting a Gradle build environment configured well, with good caching, needs to be done before Fastlane becomes useful, imho.

Here's an example that I recently added into Packagecloud, to exercise gradle push and pull from a private repository:

https://github.com/computology/packagecloud.io/tree/main/spec/fixtures/gradle/test-app

and a system test:

https://github.com/computology/packagecloud.io/blob/main/spec/system/gradle_client_spec.rb

I'd like to wrap this up into a Buildkite pipeline and get it pushing the build artifacts to the private repo [using Buildkite OIDC tokens](https://buildkite.com/docs/agent/v3/cli-oidc), too.

**Comment by olyism** [comment]
Hi @sj26 I've created a separate issue https://github.com/buildkite/templates/issues/58 to focus on the use cases you mentioned

Thanks for the inspirations :)
